src/main/java/uk/gov/ons/ssdc/supporttool/endpoint/JobEndpoint.java

```diff
@@ -146,11 +150,19 @@ public String getErrorCsv(
       throw new RuntimeException(e);
     }
 
+    if (downloadableErroredJobIds.contains(job.getId())) {
+      jobRowRepository.deleteByJobAndJobRowStatus(job, JobRowStatus.VALIDATED_ERROR);
+      downloadableErroredJobIds.remove(job.getId());
+    } else {
+      downloadableErroredJobIds.add(job.getId());
+    }
+
```

Comment:
Instead of doing it this way, I think it would be better to just run `jobRowRepository.deleteByJobAndJobRowStatus(job, JobRowStatus.VALIDATED_ERROR);` when the cancel button is ran. Just seems like it'll be better since once you press cancel that's the last thing you can do with the sample